### DISCUSSION TITLE
Error message when opening game afer New Years

### DISCUSSION THREAD

**Discussion by X-Gamerin** [Monika-After-Story/MonikaModDev#9833] [QA - tech support + general] [answered]
Can someone help me?

[code]
I'm sorry, but an uncaught exception occurred.

While running game code:
  File "game/zz_history.rpy", line 865, in script
  File "game/zz_history.rpy", line 930, in <module>
  File "game/zz_history.rpy", line 884, in _runMHSAlg
  File "game/zz_history.rpy", line 825, in save
AttributeError: 'set' object has no attribute 'iteritems'

-- Full Traceback ------------------------------------------------------------

Full traceback:
  File "game/zz_history.rpy", line 865, in script
  File "D:\Privat\MAS2\renpy\ast.py", line 814, in execute
    renpy.python.py_exec_bytecode(self.code.bytecode, self.hide, store=self.store)
  File "D:\Privat\MAS2\renpy\python.py", line 1719, in py_exec_bytecode
    exec bytecode in globals, locals
  File "game/zz_history.rpy", line 930, in <module>
  File "game/zz_history.rpy", line 884, in _runMHSAlg
  File "game/zz_history.rpy", line 825, in save
AttributeError: 'set' object has no attribute 'iteritems'

Windows-8-6.2.9200
Ren'Py 6.99.12.4.2187
Monika After Story 0.12.13-unstable.2022.12.22.166
[/code]

and also when I reload

[code]
I'm sorry, but an uncaught exception occurred.

While running game code:
  File "game/zz_history.rpy", line 865, in script
    init -800 python in mas_history:
  File "game/zz_history.rpy", line 930, in <module>
    _runMHSAlg()
  File "game/zz_history.rpy", line 884, in _runMHSAlg
    mhs.save()
  File "game/zz_history.rpy", line 825, in save
    for p_key, data_key in self.mapping.iteritems():
AttributeError: 'set' object has no attribute 'iteritems'

-- Full Traceback ------------------------------------------------------------

Full traceback:
  File "game/zz_history.rpy", line 865, in script
    init -800 python in mas_history:
  File "D:\Privat\MAS2\renpy\ast.py", line 814, in execute
    renpy.python.py_exec_bytecode(self.code.bytecode, self.hide, store=self.store)
  File "D:\Privat\MAS2\renpy\python.py", line 1719, in py_exec_bytecode
    exec bytecode in globals, locals
  File "game/zz_history.rpy", line 930, in <module>
    _runMHSAlg()
  File "game/zz_history.rpy", line 884, in _runMHSAlg
    mhs.save()
  File "game/zz_history.rpy", line 825, in save
    for p_key, data_key in self.mapping.iteritems():
AttributeError: 'set' object has no attribute 'iteritems'

Windows-8-6.2.9200
Ren'Py 6.99.12.4.2187
Monika After Story 0.12.13
[/code]

**Comment by Booplicate** [comment]
Have you ever used a translation? What submods do you have?

**Comment by X-Gamerin** [comment]
Hi! No I haven't used any translations.

![Screenshot (61)](https://user-images.githubusercontent.com/121737407/210169338-93a25b63-362f-4136-b9c8-9247f023e653.png)
![Screenshot (62)](https://user-images.githubusercontent.com/121737407/210169340-9ddb3dfe-81fc-4ea8-9b48-8544eec37326.png)
![Screenshot (63)](https://user-images.githubusercontent.com/121737407/210169341-f77f2176-2a2b-4491-9f8b-1ab2b2372800.png)
![Screenshot (64)](https://user-images.githubusercontent.com/121737407/210169342-6308a0f8-ba28-430b-9d9f-a72bde359a48.png)
![Screenshot (65)](https://user-images.githubusercontent.com/121737407/210169343-c791410d-3f6d-4d66-a2b1-4e23e7c03530.png)

and these are my submods. I know, I know, I have a ... few.

**Comment by Booplicate** [comment]
That's too many - I won't be able to narrow it down.
Show your `game/` folder.

**Comment by X-Gamerin** [comment]
![Screenshot (66)](https://user-images.githubusercontent.com/121737407/210170725-98f07f73-d4f2-44ae-b542-619d89f7f7db.png)
![Screenshot (67)](https://user-images.githubusercontent.com/121737407/210170730-a98e7e83-850e-464b-be3a-0f8869942a90.png)
![Screenshot (68)](https://user-images.githubusercontent.com/121737407/210170732-5084dc23-3068-4926-aeae-973da76b28a6.png)
![Screenshot (69)](https://user-images.githubusercontent.com/121737407/210170733-3c242947-f97d-4db3-911f-f46add8517c9.png)
![Screenshot (70)](https://user-images.githubusercontent.com/121737407/210170734-9b0064f8-ee4f-46b4-b2ce-d75d4d0c5cfb.png)
![Screenshot (71)](https://user-images.githubusercontent.com/121737407/210170735-aebc7549-5222-42e1-aef1-67deb8e2c1b0.png)
![Screenshot (72)](https://user-images.githubusercontent.com/121737407/210170737-c1c1c00f-a6a9-4cda-bf2e-3d3771b11a88.png)
![Screenshot (73)](https://user-images.githubusercontent.com/121737407/210170738-4c38dcd9-4f8c-487d-b0aa-33cdbdf3ef0d.png)


here it is

**Comment by Booplicate** [comment]
You're using source version, this is not supported.
[Download fresh copy of DDLC](https://ddlc.moe/), [download MAS installer](https://monikaafterstory.com/releases.html), and reinstall.

**Comment by Booplicate** [comment, marked as answer]
This is also a submod issue, `MSautism.rpy` is causing this and doesn't even have submod header. Uninstall it and restore a backup from before you started using it.
